### PR TITLE
fix(cluster_k8s): check if cert-manager is actually operational

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/Hj70Xle2/2712-k8s-address-cert-manager-issue

https://cloudius-jenkins-test.s3.amazonaws.com/8e05f3e5-94a2-40e2-a318-3515c41ad4eb/20201214_090614/sct-runner-8e05f3e5.zip

Addressed issue when build fails due to the no operational cert-manager:

```
2020-12-14 09:05:56.914: (TestFrameworkEvent Severity.ERROR), source=LongevityTest.SetUp()
exception=Encountered a bad command exit code!
Command: 'kubectl apply -f <(envsubst</jenkins/slave/workspace/operator-master/longevity-scylla-operator-3h-gke-cluster-rolling/scylla-cluster-tests/sdcm/k8s_configs/operator.yaml)'
Exit code: 1
Stdout:
namespace/scylla-operator-system created
customresourcedefinition.apiextensions.k8s.io/scyllaclusters.scylla.scylladb.com created
mutatingwebhookconfiguration.admissionregistration.k8s.io/scylla-operator-mutating-webhook-configuration created
clusterrole.rbac.authorization.k8s.io/scylla-operator-manager-role created
clusterrolebinding.rbac.authorization.k8s.io/scylla-operator-manager-rolebinding created
service/scylla-operator-webhook-service created
statefulset.apps/scylla-operator-controller-manager created
issuer.cert-manager.io/scylla-operator-selfsigned-issuer created
validatingwebhookconfiguration.admissionregistration.k8s.io/scylla-operator-validating-webhook-configuration created
Stderr:
Error from server (InternalError): error when creating "/dev/fd/63": Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=30s: x509: certificate signed by unknown authority
```

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster_k8s/__init__.py [hunk 1]
```diff
@@ -108,8 +109,26 @@ def deploy_cert_manager(self) -> None:
...
+        # In this function we use dirty way to estimate that by deploying scylla-operator
+        # which invokes cert-manager hooks and making sure that next step after cert-manager deploying,
+        # which is scylla-operator deploying, won't fail.
+        try:
+            self.apply_file(SCYLLA_OPERATOR_CONFIG)
+            return True
+        finally:
+            self.kubectl('delete namespace scylla-operator', ignore_status=True)
+        return False
+
```
Comment (enaydanov) [inline sdcm/cluster_k8s/__init__.py:131]: It's too dirty :) Can we use an example from https://cert-manager.io/docs/installation/kubernetes/#verifying-the-installation ?

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/cluster_k8s/__init__.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/cluster_k8s/__init__.py]: fixed

File: sdcm/cluster_k8s/__init__.py [hunk 2]
```diff
@@ -108,8 +110,26 @@ def deploy_cert_manager(self) -> None:
...
+            timeout=10 * 60,
+            step=10,
+            throw_exc=True)
         self.start_cert_manager_journal_thread()
 
+    def check_if_cert_manager_fully_functional(self):
+        # It is not possible to say with 100% certainty that cert-manager become fully operational
+        # In this function we use dirty way to estimate that by deploying scylla-operator
+        # which invokes cert-manager hooks and making sure that next step after cert-manager deploying,
+        # which is scylla-operator deploying, won't fail.
```
Comment (enaydanov) [inline sdcm/cluster_k8s/__init__.py:125]: Outdated comment.

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/cluster_k8s/__init__.py [hunk 3]
```diff
@@ -108,8 +110,26 @@ def deploy_cert_manager(self) -> None:
...
+        try:
+            self.apply_file(CERT_MANAGER_TEST_CONFIG)
+            return True
```
Comment (enaydanov) [inline sdcm/cluster_k8s/__init__.py:128]: Can you add a check or just debug output (`kubectl describe certificate -n cert-manager-test`)

File: sdcm/cluster_k8s/__init__.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/cluster_k8s/__init__.py:125]: commed fixed

File: sdcm/cluster_k8s/__init__.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/cluster_k8s/__init__.py:126]: kubectl apply is pretty verbose, why do we need it ?

**Review by bentsi — APPROVED** [review]

**Pull request merged**